test/smart-wallet/signatures/EIP1271Test.ts

```diff
@@ -0,0 +1,589 @@
+    let mainSmartAccount2Address = await walletFactory.getAddressForCounterfactualWallet(signerSmartAccount2Address, deployWalletIndex);
+    
+    await walletFactory.deployCounterFactualWallet(
+      signerSmartAccount2Address,
+      deployWalletIndex
+    );
+
+    let mainSmartAccount2 = await ethers.getContractAt(
+      "contracts/smart-contract-wallet/SmartAccount.sol:SmartAccount",
+      mainSmartAccount2Address
+    );
+    
+    await token
+      .connect(accounts[0])
+      .transfer(mainSmartAccount.address, ethers.utils.parseEther("100"));
+    
+    await token
+      .connect(accounts[0])
+      .transfer(signerSmartAccount.address, ethers.utils.parseEther("100"));
+
+    await token
+      .connect(accounts[0])
+      .transfer(signerSmartAccount2.address, ethers.utils.parseEther("100"));
+
+    await token
+      .connect(accounts[0])
+      .transfer(mainSmartAccount2.address, ethers.utils.parseEther("100"));
+
+    expect(await token.balanceOf(charlie)).to.equal(0);
+
+    let tokensToBeTransferred = ethers.utils.parseEther("10");
+
+    // TX TO TRANSFER 10 tokens FROM mainSmartAccount to Charlie 
+    const safeTx: SafeTransaction = buildSafeTransaction({
+      to: token.address,
+      data: encodeTransfer(charlie, tokensToBeTransferred.toString()),
+      nonce: await mainSmartAccount.getNonce(EOA_CONTROLLED_FLOW),
+    });
+
+    const transaction: Transaction = {
+      to: safeTx.to,
+      value: safeTx.value,
+      data: safeTx.data,
+      operation: safeTx.operation,
+      targetTxGas: safeTx.targetTxGas,
+    };
+    const refundInfo: FeeRefund = {
+      baseGas: safeTx.baseGas,
+      gasPrice: safeTx.gasPrice,
+      tokenGasPriceFactor: safeTx.tokenGasPriceFactor,
+      gasToken: safeTx.gasToken,
+      refundReceiver: safeTx.refundReceiver,
+    };
+
+    const chainId = await mainSmartAccount.getChainId();
+
+    // BUILD 1271 SIGNATURE BY SIGNER SMART ACCOUNT
+
+    const { signer, data } = await safeSignTypedData(
+      accounts[0], //owner
+      mainSmartAccount,
+      safeTx,
+      chainId
+    );
+    
+    let signature = buildContractSignature(signerSmartAccount.address, data);
+
+    // MANIPULATE SIGNATURE TO SET signerContract2 as verifier
+    let addressToInsert = signerSmartAccount2.address.slice(2);
+
+    let manipulatedSignature = signature.replace(signature.substring(26,66), addressToInsert);
+
+    // Expect can not use this signature on main smart account 2, even despite 
+    // it is owned by signer smart account 2, that is owned by the owner (original signer)
+    await expect(
+      mainSmartAccount2.connect(accounts[1]).execTransaction(
+        transaction,
+        refundInfo,
+        manipulatedSignature
+      )
+    ).to.be.revertedWith("BSA024");
+
+    expect(await token.balanceOf(charlie)).to.equal(0);
+
+  });
+
+  
+it("0x exploit 1271 | Reverts if trying to use 1271 signature instead of EOA signature", async function () { 
```

Comment:
I guess we don't need to make change like this commit? https://github.com/0xProject/0x-monorepo/commit/2171632e1e0f83ca55d7cc0072e3fb6d658219a0

Reply:
I think , we don't. as tests show that isValidSignature calls to eoa revert